src/main/java/org/folio/rest/impl/InstanceStorageAPI.java

```diff
@@ -144,11 +152,23 @@ public void postInstanceStorageInstances(
               reply -> {
                 try {
                   if(reply.succeeded()) {
-                    asyncResultHandler.handle(
-                      io.vertx.core.Future.succeededFuture(
-                        PostInstanceStorageInstancesResponse
-                          .respond201WithApplicationJson(entity,
-                              PostInstanceStorageInstancesResponse.headersFor201().withLocation(reply.result()))));
+                    domainEventService.instanceCreated(entity)
+                    .whenComplete((notUsed, error) -> {
+                      if (error == null) {
+                        asyncResultHandler.handle(
+                          io.vertx.core.Future.succeededFuture(
+                            PostInstanceStorageInstancesResponse
+                              .respond201WithApplicationJson(entity,
+                                PostInstanceStorageInstancesResponse.headersFor201()
+                                  .withLocation(reply.result()))));
```

Comment:
Does this mean that it has been decided that if publishing to Kafka fails (and hence any consumers will become inconsistent) the API operation should still succeed?

Reply:
No, we agreed that if kafka message failed to be published then the failure should be sent back to the client.

Reply:
@bohdan-suprun Thanks, I misread the code.

Does this mean that we will communicate to the client that the update failed, yet the database state has been changed?

Reply:
Unfortunately yes.

Reply:
I think it might make make sense to document that as a design decision.

@MikhailFokanov what do you think?